src/NuGet.Core/NuGet.ProjectModel/DependencyGraphSpec.cs

```diff
@@ -137,6 +137,39 @@ public DependencyGraphSpec WithProjectClosure(string projectUniqueName)
             return projectDependencyGraphSpec;
         }
 
+        /// <summary>
+        /// Creates a new <see cref="DependencyGraphSpec" /> from a project name and project closure.
+        /// </summary>
+        /// <param name="projectUniqueName">The project's unique name.</param>
+        /// <param name="closure">The project's closure</param>
+        /// <returns>A <see cref="DependencyGraphSpec" />.</returns>
+        /// <exception cref="ArgumentException">Thrown if <paramref name="projectUniqueName" />
+        /// is <c>null</c> or an empty string.</exception>
+        /// <exception cref="ArgumentNullException">Thrown if <paramref name="closure" /> is <c>null</c>.</exception>
+        public DependencyGraphSpec CreateFromClosure(string projectUniqueName, IReadOnlyList<PackageSpec> closure)
+        {
+            if (string.IsNullOrEmpty(projectUniqueName))
```

Comment:
Go ahead and check for whitespace as well? 
Should 1 space versus 2 or more spaces be considered distinct names? (eg, " " versus "     " should be unique?)
Should "myProject" be distinct from "myProject " (ie, should we Trim spaces?)

Reply:
I agree with @zivkan.  There's a spectrum of possible validations.  We should avoid no validation.  Basic input validation is sufficient, but "basic" is open to interpretation.  I think `null` and zero-length strings are easy, safe validations; it's a small set of disallowed values.  I'd prefer in this case to leave it to higher level consumers to enforce whether or not whitespaces are allowed.